lib/runnable.js

```diff
@@ -81,10 +80,6 @@ Runnable.prototype.timeout = function(ms) {
   var range = [0, INT_MAX];
   ms = utils.clamp(ms, range);
 
-  // see #1652 for reasoning
-  if (ms === range[0] || ms === range[1]) {
-    this._enableTimeouts = false;
```

Comment:
Here we check wether the value is inside of a certain range?
If not, shouldn't we set `this._timeout = 0` ?

Reply:
Docs:
> a parameter passed to `this.timeout()` greater than the maximum delay value will cause the timeout to be disabled.

Yes you are right.